### PR TITLE
Added Kindergarten Garden

### PR DESCRIPTION
**Pull request by ryanplusplus**
See: https://github.com/exercism/x-common/blob/master/kindergarten-garden.md

### CODE REVIEW

File: kindergarten-garden/kindergarten-garden_spec.lua [hunk 1]
```diff
@@ -0,0 +1,52 @@
+local Garden = require('kindergarten-garden')
+
+describe('kindergarten garden', function()
+
+  it('should be able to parse a garden with only one student', function()
+    local garden = Garden('RC\nGG')
+    assert.same({ 'radishes', 'clover', 'grass', 'grass' }, garden.alice)
```
Comment (ryanplusplus) [inline kindergarten-garden/kindergarten-garden_spec.lua]: I chose to index the object with the student name to reward the use of metatables

File: kindergarten-garden/example.lua [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+return function(s)
+  s = s:lower()
+
+  return setmetatable({}, {
+    __index = function(_, student)
+      local result = {}
+      plant_string_for_student(s, student:lower()):gsub('.', function(c)
+        table.insert(result, plant_abbreviations[c])
+      end)
```
Comment (aarti) [inline kindergarten-garden/example.lua]: Would be easier to read if the anonymous function that did the split and map was a separate local function.

File: kindergarten-garden/kindergarten-garden_spec.lua (hunk 1 above)
Comment (aarti) [inline kindergarten-garden/kindergarten-garden_spec.lua]: Looks good, metatables are important.

File: kindergarten-garden/example.lua [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  joseph = 9,
+  kincaid = 10,
+  larry = 11
+}
+
+local plant_abbreviations = { g = 'grass', c = 'clover', r = 'radishes', v = 'violets' }
+
+local function plant_string_for_student(garden, student)
+  if not student_offsets[student] then return '' end
+  local _, _, row1, row2 = garden:find('(%w+)\n(%w+)')
```
Comment (aarti) [inline kindergarten-garden/example.lua]: Nice split row regex :+1:

**Comment by aarti** [comment]
Looks good

**Pull request merged**